[댓글 기능] 무민 미션 제출합니다.

green4469 opened the pull request:
많은 지적 부탁드립니다. :D

green4469 committed:
refactor: return boolean expression directly

green4469 committed:
refactor: use HttpStatus constants instead of magic numbers for HTTP status code

green4469 committed:
refactor: separate packages for dtos, exceptions in service layer

green4469 committed:
refactor: rename controller methods in ArticleController

green4469 committed:
refactor: add update method for article

green4469 committed:
refactor: update article-edit.html article update url

green4469 committed:
chore: update properties file

green4469 committed:
chore: add data initialization file for test

green4469 committed:
refactor: simplify article controller test

green4469 committed:
feat: add oneToMay bidirectional relationship

green4469 committed:
feat: add Comment Entity

green4469 committed:
feat: add createdAt column into Comment

green4469 committed:
feat: add CommentRepository

green4469 committed:
feat: create/read comments

green4469 committed:
feat: delete comment

green4469 committed:
debug: use transactional annotations on update services

green4469 committed:
feat: update comment

green4469 committed:
feat: delete comments when delete its parent article

green4469 committed:
feat: add modify comment frontend

woowahanCU reviewed (changes requested):
안녕하세요. 리뷰어 CU입니다.
빠른 시간내에 잘 구현하셨네요! 👍 
댓글에 대한 테스트코드도 작성하셨네요 ㅎㅎ
몇가지 피드백 남겨두었으니 확인해보시고 궁금한 점 있으면 DM 남겨주세요.

green4469 committed:
refactor: remove unnecessary setters, constructor in Comment class

green4469 committed:
refactor: remove comments

green4469 committed:
refactor: remove articleRepository field from ArticleController

green4469 committed:
refactor: remove commentRepository field from ArticleController

green4469 committed:
refactor: use repository's return object of 'save' method as persistence entity

green4469 committed:
refactor: use "deleteByArticle"

green4469 committed:
refactor: add dto collection, simplify controller

green4469 committed:
refactor: debug test code; missing session id header

green4469 committed:
refactor: add ON_DELETE option to comment -> article reference

green4469 committed:
refactor: add ON_DELETE option to comment -> article reference

green4469 committed:
refactor: active data.sql in test

green4469 committed:
refactor: use "import.sql"

green4469 committed:
refactor: use "data.sql" and TestPropertySource

green4469 committed:
refactor: remove dto-creation logic from controller

green4469 committed:
refactor: change status code for invalid form from 400 into 422

green4469 committed:
refactor: add length limit to string fields in entities

green4469 committed:
refactor: use ApplicationRunner instead of data.sql to initialize database

green4469 committed:
refactor: use @EqualsAndHashcode only for id, not for mutable fields

Hashcode can change when value of fields changes. (https://kwonnam.pe.kr/wiki/java/lombok/pitfall)

green4469 committed:
refactor: use Spring JPA Auditing to handle created date of entity 'Comment'

green4469 committed:
refactor: change url path ("/writing" -> "/articles/writing")

green4469 committed:
refactor: add VOs, remove BodyInserter duplication in ArticleController test.

woowahanCU reviewed (commented):
https://github.com/woowacourse/jwp-blog/pull/107#issuecomment-517576146

같은 이슈가 있네요.

woowahanCU reviewed (changes requested):
질문하신 부분 몇가지 피드백 남겨두었습니다.
궁금한 부분 있으시면 DM 남겨주세요~

green4469 committed:
refactor: remove unnecessary declaration in ArticleRepository

green4469 committed:
refactor: return unmodifiable list in services

green4469 committed:
refactor: remove url path duplication in CommentController

green4469 committed:
feat: calculate the minutes ago when comment is written, how many comments are written

green4469 committed:
feat: use @URL validator to Article's cover url

green4469 committed:
refactor: Entities don't depend on dtos anymore.

green4469 committed:
refactor: removing duplication in ArticleControllerTest

green4469 committed:
refactor: add put, delete helper methods

woowahanCU reviewed (approved):
피드백 반영 정말 잘 해주셨네요! 💯 
미션 진행하느라 고생 많았어요. ㅎㅎ 
이번 미션이 많은 도움 되었길 바래요~ 😀
